internal/pkg/config/config.go

```diff
@@ -123,21 +141,27 @@ func (c Config) IsSourceField(typ types.Type, fld *types.Var) bool {
 }
 
 func (c Config) IsSourceFieldAddr(fa *ssa.FieldAddr) bool {
```

Comment:
A docstring might be helpful here now that the complexity has grown.

If I'm reading this correctly, the intent is:
* Check if the accessed field itself is a source type.  If so, return `true`.
* Check if the wrapping struct containing the field is a source type.  If so, check that the accessed field is marked.

It could be part of a future PR, but given the fact that we're matching based on names here, we're going to have a problem for fields annotated as sources by a field tag.

Reply:
Agreed. In fact given the complexity I think it would be useful to add unit tests and refactor, but that would definitely need its own PR.

Concerning field tags, I think it could be done in this function (that would increase the complexity a bit more, but to me it feels like the right place to put it).